packages/jellyfish-api-core/__tests__/category/spv/createAnchor.test.ts

```diff
@@ -0,0 +1,242 @@
+      txid: 'INVALID',
+      vout: 2,
+      amount: 15800,
+      privkey: 'b0528d87cfdb09f72c9d10b7b3cc00727062d93537a3e8abcf1fde821d08b59d'
+    }], rewardAddress)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('txid must be of length 64 (not 7, for \'INVALID\')')
+  })
+
+  it('should be failed as not enough money to create anchor', async () => {
+    const rewardAddress = await clients[0].spv.getNewAddress()
+
+    const promise = clients[0].spv.createAnchor([{
+      txid: '11a276bb25585f6973a4dd68373cffff41dbcaddf12bbc1c2b489d1dc84564ee',
+      vout: 2,
+      amount: 1,
+      privkey: 'b0528d87cfdb09f72c9d10b7b3cc00727062d93537a3e8abcf1fde821d08b59d'
+    }], rewardAddress)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Not enough money to create anchor') // Not enough money to create anchor: 1 (need 3522)
+  })
+
+  it('should be failed as invalid privkey', async () => {
+    const rewardAddress = await clients[0].spv.getNewAddress()
+
+    const promise = clients[0].spv.createAnchor([{
+      txid: '11a276bb25585f6973a4dd68373cffff41dbcaddf12bbc1c2b489d1dc84564ee',
+      vout: 2,
+      amount: 15800,
+      privkey: 'INVALID'
+    }], rewardAddress)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('spv: Can\'t parse WIF privkey INVALID')
+  })
+
+  it('should be failed as invalid address', async () => {
+    const promise = clients[0].spv.createAnchor([{
+      txid: '11a276bb25585f6973a4dd68373cffff41dbcaddf12bbc1c2b489d1dc84564ee',
+      vout: 2,
+      amount: 15800,
+      privkey: 'b0528d87cfdb09f72c9d10b7b3cc00727062d93537a3e8abcf1fde821d08b59d'
+    }], 'INVALID')
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('rewardAddress (INVALID) does not refer to a P2PKH or P2WPKH address')
+  })
+
+  it('should createAnchor', async () => {
+    const rewardAddress = await clients[0].spv.getNewAddress()
+
+    const anchor = await clients[0].spv.createAnchor([{
+      txid: '11a276bb25585f6973a4dd68373cffff41dbcaddf12bbc1c2b489d1dc84564ee',
+      vout: 2,
+      amount: 15800,
+      privkey: 'b0528d87cfdb09f72c9d10b7b3cc00727062d93537a3e8abcf1fde821d08b59d'
+    }], rewardAddress)
+    expect(typeof anchor.txHex).toStrictEqual('string')
+    expect(typeof anchor.txHash).toStrictEqual('string')
+    expect(typeof anchor.defiHash).toStrictEqual('string')
+    expect(anchor.defiHeight).toStrictEqual(15)
+    expect(anchor.estimatedReward).toStrictEqual(new BigNumber(0))
+    expect(anchor.cost).toStrictEqual(new BigNumber(3556))
+    expect(anchor.sendResult).toStrictEqual(0)
+    expect(anchor.sendMessage).toStrictEqual('Success')
+
+    {
+      const pending = await group.get(0).call('spv_listanchorspending')
+      // pending does not contain active property
+      expect(pending.active).toBeUndefined()
```

Comment:
pending does not have `active` property anyways. so maybe instead check the length of the `pending` object?